modules/authentication/components/ForgotPassword.jsx

```diff
@@ -0,0 +1,66 @@
+import React, {
+  cloneElement,
+  Component,
+  PropTypes
+} from 'react';
+
+const propTypes = {
+  actions: PropTypes.shape({
+    authentication: PropTypes.shape({
+      requestPasswordReset: PropTypes.func.isRequired
+    }).isRequired
+  }).isRequired,
+  children: PropTypes.node,
+  location: PropTypes.shape({
+    query: PropTypes.shape({
+      next: PropTypes.string
+    }).isRequired
+  }).isRequired,
+  router: PropTypes.shape({
+    replace: PropTypes.func.isRequired
+  })
+};
+
+class ForgotPassword extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+
+    const values = Array.prototype.reduce.call(this.forgotPasswordForm.elements, (memo, element) => {
+      if (element.name && element.value) {
+        memo[element.name] = element.value;
+      }
+
+      return memo;
+    }, {});
+    values.redirect_url = window.location.origin;
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLHyperlinkElementUtils/origin#Browser_compatibility

^^ you need to build it yourself out of the host and port and protocol ^^